enhancements/user-workload-monitoring.md

```diff
@@ -0,0 +1,275 @@
+---
+title: user-workload-monitoring
+authors:
+  - "@s-urbaniak"
+  - "@brancz"
+reviewers:
+  - "@sichvoge"
+  - "@openshift/openshift-team-monitoring"
+approvers:
+  - "@sichvoge"
+  - "@openshift/openshift-team-monitoring"
+creation-date: 2019-09-19
+last-updated: 2019-09-19
+status: implementable
+---
+
+# user-workload-monitoring
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+This enhancement describes a new feature for OpenShift cluster observability, namely user workload monitoring.
+User workload monitoring will not be enabled by default, it is an opt-in feature
+that can be enabled after installation.
+It is an extension to the existing cluster monitoring stack to enable observability of user namespaces.
+Currently, the cluster monitoring stack only reconciles OpenShift namespaces.
+
+## Motivation
+
+Currently, there is no supported solution for customers to monitor their business critical applications.
+The existing cluster monitoring stack has no support for adding scrape targets in user namespaces
+as it is only available for OpenShift components scraping workloads from `openshift-*` namespaces.
+Other Red Hat products, outside the OpenShift framework components also
+don’t have a monitoring solution to integrate their metrics and out-of-the-box insights into their well being.
+Currently it is difficult to achieve metering/chargeback for Red Hat products without a central observability stack.
+
+### Goals
+
+Allow deployment and reconciliation of scrape targets, recording rules, and alerting rules in user namespaces.
+This effectively means the following custom resources:
+
+-  [ServiceMonitor](https://github.com/openshift/prometheus-operator/blob/master/Documentation/api.md#servicemonitor)
+- [PodMonitor](https://github.com/openshift/prometheus-operator/blob/master/Documentation/api.md#podmonitor)
+- [PrometheusRule](https://github.com/openshift/prometheus-operator/blob/master/Documentation/api.md#prometheusrule)
+
+Allow the correlation between app and cluster centric metrics.
+
+### Non-Goals
+
+Out of scope are:
+
+- Definition of Dashboards.
+- Deployment of Prometheus instances in user namespaces.
+- Deployment of Alertmanager instances in user namespaces.
+
+## Proposal
+
+![](user-monitoring-deployment.png)
+
+The existing cluster monitoring stack is going to be extended with additional Prometheus (P) servers
```

Comment:
Is there a fixed number of Prometheus instances (eg 2) that will be deployed?

Reply:
Yes, it will be the same model as cluster monitoring and will probably start with 2.